content/ros/pubsub/pubsub.tex

```diff
@@ -0,0 +1,123 @@
+
+\documentclass{beamer}
+\usepackage{changepage}
+\usepackage{lmodern}
+\usetheme{default}
+\usepackage{listings}
+
+\title{ROS Publisher/Subscriber Model}
+
+\author{Kiran Vasudev\inst{1}}
+
+\institute[] 
+{
+  \inst{1}
+  Hochschule Bonn-Rhein-Sieg
+
+}
+
+\begin{document}
+
+\begin{frame}
+  \titlepage
+\end{frame}
+
+\begin{frame}{Outline}
+  \tableofcontents
+\end{frame}
+
+\begin{frame}{The Publisher/Subscriber Model}
+\section{The Publisher/Subscriber Model}
+  \begin{itemize}
+  \item {
+	Transport system used to route messages
+  }
+  \item {
+	A node sends a message by publishing it to a topic
+  }
+  \item {A node that is interested to access/use data certain data will subscribe to the most appropriate topic}
+  \item{There can be many existing publishers and subscribers for a single topic}
+  \item{A single node can publish/subscribe to multiple topics}
+  \end{itemize}
+\end{frame}
+
+\begin{frame}
+	\begin{figure}
+		\begin{center}
+			\includegraphics[scale=0.8]{images/basic_ros}
+			\caption{The publish/subscribe model [1]}	
+		\end{center}
+	\end{figure}
+\end{frame}
+
+
+\begin{frame}{Cons of the Publisher/Subscriber Model}
+\section{Cons of the Publisher/Subscriber Model}
+  \begin{itemize}
+  \item {
+    The model's many-to-many one-way transport of messages are not appropriate for interactions based on requests/replies.
+  }
+  \item {   
+    This con is overcome by using services (Will be introduced in the next session)
+  }	
+  \end{itemize}
+
+\end{frame}
+
+\begin{frame}[fragile]{Important commands}
+\section{Important commands}
+\begin{itemize}
+	\item {Python\\
+	\begin{itemize}
+		\item{\textbf{Publisher}}\\
+		\footnotesize{pub = rospy.Publisher('chatter', String, queue\_size=10)}	\\
+		\footnotesize{pub.publish(msg)}	\\
+		
+		\item{\textbf{Subscriber}}\\
+		\footnotesize{rospy.Subscriber("chatter", String, callback)}	
+	\end{itemize}
+	}
+
+	\item {C++ \\	
+	\begin{itemize}
+		\item{\textbf{Publisher}}\\
+		\footnotesize{ros::Publisher chatter\_pub = n.advertise$\langle$std\_msgs::String $\rangle$("chatter", 1000);}	\\
+		\footnotesize{chatter\_pub.publish(msg);}
+		
+		\item{\textbf{Subscriber}}\\
+		\footnotesize{ros::Subscriber sub = n.subscribe("chatter", 1000, chatterCallback);}	
+	\end{itemize}
+}
+\end{itemize}
+\end{frame}	
+
+\begin{frame}{Task}
+	\section{Task}
+	\begin{itemize}
+		\item Create a new ROS package that has a minimum of two nodes: A publisher and a subscriber
+		\item Define a custom message that contains:
+		\begin{itemize}
+			\item Header
```

Comment:
I would add another custom thing that has not been predefined in existing ROS messages, to make it a bit clearer.